src/release/breaking-changes/android-14-nonlinear-text-scaling.md

```diff
@@ -0,0 +1,292 @@
+
+Before:
+```dart 
+UnmigratedTextWidget( 
+  textScaleFactor: MediaQuery.textScaleFactorOf(context),
+  ...
+)
+```
+
+After:
+```dart 
+UnmigratedTextWidget( 
+  textScaleFactor: MediaQuery.textScalerOf(context).textScaleFactor,
+  ...
+)
+```
+
+If you wish to compute the scaled font size yourself, use `TextScaler.scale` 
+instead of the `*` binary operator:
+
+Before:
+```dart 
+final scaledFontSize = textStyle.fontSize * MediaQuery.textScaleFactorOf(context);
+```
+
+After:
+```dart 
+final scaledFontSize = MediaQuery.textScalerOf(context).scale(textStyle.fontSize);
+```
+
+
+If the API that provides `textScaleFactor` hasn't been migrated, consider 
+waiting for the migrated version.
+
+If you are using `textScaleFactor` to scale dimensions that are not font sizes, 
+there are no generic rules for migrating the code to nonlinear scaling, and it 
+might require the UI to be implemented differently. Reusing the `MyTooltipBox` 
+example:
+```dart 
+MyTooltipBox( 
+  size: chatBoxSize * textScaleFactor,
+  child: RichText(..., style: TextStyle(fontSize: 20)),
+)
+```
+You could choose to use the "effective" text scale factor by applying the 
+`TextScaler` on the font size 20: `chatBoxSize * textScaler.scale(20) / 20`, or
+redesign the UI and let the widget assume its own intrinsic size.
+
+### Overriding the text scaling strategy in a widget subtree
+
+To override the existing `TextScaler` used in a widget subtree, override the 
+`MediaQuery` like so:
+
+Before:
+```dart
+MediaQuery( 
+  data: MediaQuery.of(context).copyWith(textScaleFactor: 2.0),
+  child: child,
+)
+```
+
+After:
+```dart
+MediaQuery( 
+  data: MediaQuery.of(context).copyWith(textScaler: _myCustomTextScaler),
+  child: child,
+)
+```
+
+However it's rarely needed to create a custom `TextScaler` subclass.
+`MediaQuery.withNoTextScaling` (which creates a widget that disables text scaling 
+altogether for its child subtree), and `MediaQuery.withClampedTextScaling` (which 
+creates a widget that restricts the scaled font size to within the range 
+`[minScaleFactor * fontSize, maxScaleFactor * fontSize]`), are convenience methods 
+that cover common cases where the text scaling strategy needs to be overridden. 
+
+#### Examples
+
+**Disabling Text Scaling For Icon Fonts**
+Before:
+```dart 
+MediaQuery(
+  data: MediaQuery.of(context).copyWith(textScaleFactor: 1.0),
+  child: IconTheme(
+    data: ..,
+    child: icon,
+  ),
+)
+```
+
+After 
+```dart 
+MediaQuery.withNoTextScaling(
+  child: IconTheme(
+    data: ...
+    child: icon,
+  ),
+)
+```
+
+**Preventing Contents From Overscaling**
+Before:
+```dart 
+final mediaQueryData = MediaQuery.of(context);
+MediaQuery(
+  data: mediaQueryData.copyWith(textScaleFactor: math.min(mediaQueryData.textScaleFactor, _kMaxTitleTextScaleFactor),
+  child: child,
+)
+```
+
+After
+```dart 
+MediaQuery.withClampedTextScaling(
+  maxScaleFactor: _kMaxTitleTextScaleFactor,
+  child: title,
+)
+```
+
+## Timeline
+
+Landed in version: xxx<br>
```

Comment:
I can't land this PR until the "Landed in" field is filled out.

Reply:
I'm planning to land the PR on thursday.